src/systems/filecoin_blockchain/storage_power_consensus/_index.md

```diff
@@ -55,38 +55,47 @@ Whenever comparing tickets is evoked in Filecoin, for instance when discussing s
 While each Filecoin block header contains a ticket field (see {{<sref tickets>}}), it is useful to think of a ticket chain abstraction.
 Due to the nature of Filecoin's Tipsets and the possibility of using tickets from epochs that did not yield leaders to produce randomness at a given epoch, tracking the canonical ticket of a subchain at a given height can be arduous to reason about in terms of blocks. To that end, it is helpful to create a ticket chain abstraction made up of only those tickets to be used for randomness generation at a given height.
 
+To read more about specifically how tickets are processed for randomness, see {{<sref randomness>}}. Note that the serialization of the ticket output (bytes) should be used directly, unlike with more complex objects where one would use the entire object serialization.
+
 To sample a ticket for a given epoch n:
-- Set referenceTipsetOffset = 0
-- While true:
-    - Set referenceTipsetHeight = n - referenceTipsetOffset
-    - If blocks were mined at referenceTipsetHeight:
-        - ReferenceTipset = TipsetAtHeight(referenceTipsetHeight)
-        - Select the block in ReferenceTipset with the smallest final ticket, return its ticket (pastTicket).
-    - If no blocks were mined at referenceTipsetHeight:
-        - Increment referenceTipsetOffset
-        - (Repeat)
-- newRandomness = H(pastTicket || minerAddress || n)
+```text
+Set referenceTipsetOffset = 0
+While true:
+    Set referenceTipsetHeight = n - referenceTipsetOffset
+    If blocks were mined at referenceTipsetHeight:
+        ReferenceTipset = TipsetAtHeight(referenceTipsetHeight)
+        Select the block in ReferenceTipset with the smallest final ticket, return its value (pastTicket).
+    If no blocks were mined at referenceTipsetHeight:
+        Increment referenceTipsetOffset
+        (Repeat)
+newRandomness = H(TicketDrawDST || epoch || Serialization(pastTicketOutput))
```

Comment:
Would prefer to use `H(DST || index || Serialization(epoch || pastTicketOutput))` where index = -1, as above, to be conceptually consistent -- not critical though.

Reply:
So when would you ever use the index?

Reply:
If you want to derive a longer pseudorandom string from a single input -- e.g. if you need 768 bits generated from `struct { epoch, ticketOutput }`, you would concatenate the results with indices (0, 1, 2).